crates/astria-composer/tests/blackbox/composer.rs

```diff
@@ -55,9 +71,69 @@ async fn tx_from_two_rollups_are_received_by_sequencer() {
         .expect("mocked sequencer should have received a broadcast message from composer");
 }
 
-/// Deserizalizes the bytes contained in a `tx_sync::Request` to a
-/// signed sequencer transaction and verifies that the contained
-/// sequence action is for the given `expected_chain_id`.
+#[tokio::test]
+async fn invalid_nonce_failure_causes_tx_resubmission_under_different_nonce() {
+    use crate::helper::mock_sequencer::mount_abci_query_mock;
+
+    // Spawn a composer with a mock sequencer and a mock rollup node
+    // Initial nonce is 0
+    let test_composer = spawn_composer(&["test1"]).await;
+    tokio::time::timeout(
+        Duration::from_millis(100),
+        test_composer.setup_guard.wait_until_satisfied(),
+    )
+    .await
+    .expect("setup guard failed");
+
+    // Reject the first transaction for invalid nonce
+    let invalid_nonce_guard =
+        mount_broadcast_tx_sync_invalid_nonce_mock(&test_composer.sequencer, "test1").await;
+
+    // Mount a response of 0 to a nonce query
+    let nonce_refetch_guard = mount_abci_query_mock(
+        &test_composer.sequencer,
+        "accounts/nonce",
+        NonceResponse {
+            height: 0,
+            nonce: 1,
+        },
+    )
+    .await;
+
+    // Expect nonce 0 again so that the resubmitted tx is accepted
+    let valid_nonce_guard =
+        mount_broadcast_tx_sync_mock(&test_composer.sequencer, "test1", 1).await;
+
+    // Push a tx to the rollup node so that it is picked up by the composer and submitted with the
+    // stored nonce of 42, triggering the nonce refetch process
```

Comment:
The above comment mentions 0, and before that a nonce response with 1 is mounte


Where does 42 come from?